azurerm/internal/services/firewall/firewall_policy_resource.go

```diff
@@ -211,6 +224,13 @@ func resourceFirewallPolicyCreateUpdate(d *pluginsdk.ResourceData, meta interfac
 		}
 	}
 
+	if v, ok := d.GetOk("private_ip_ranges"); ok {
+		privateIpRanges := utils.ExpandStringSlice(v.([]interface{}))
+		if len(*privateIpRanges) != 0 {
+			props.FirewallPolicyPropertiesFormat.Snat = &network.FirewallPolicySNAT{PrivateRanges: privateIpRanges}
+		}
```

Comment:
if we do this it's not possible to remove this, so we'll need to make this:

```suggestion
		props.FirewallPolicyPropertiesFormat.Snat = &network.FirewallPolicySNAT{
			PrivateRanges: privateIpRanges
		}
```